sdk/dotnet/Pulumi/Config_Exceptions.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright 2016-2018, Pulumi Corporation
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json.Linq;
+
+namespace Pulumi
+{
+    public partial class Config
+    {
+        /// <summary>
+        /// ConfigTypeException is used when a configuration value is of the wrong type.
+        /// </summary>
+        private class ConfigTypeException : RunException

```

Comment:
Nit: While not strictly necessary, these private exception types could all be sealed.

Reply:
Yeah, def don'tcare :)